.github/workflows/rmc.yml

```diff
@@ -46,3 +46,16 @@ jobs:
 
       - name: Execute RMC regression
         run: ./scripts/rmc-regression.sh
+
+      # On one OS only, build the documentation, too.
+      - name: Build Documentation
+        if: ${{ matrix.os == 'ubuntu-20.04' }}
+        run: ./rmc-docs/build-docs.sh
+
+      # When we're pushed to main branch, only then actually publish the docs.
+      - name: Publish Documentation
+        if: ${{ matrix.os == 'ubuntu-20.04' && github.event_name == 'push' && startsWith('refs/heads/main', github.ref) }}
+        uses: JamesIves/github-pages-deploy-action@4.1.4
+        with:
+          branch: gh-pages
```

Comment:
Is there any other way to publish github doc? In my previous team, this branch blew up once people added images to the doc. Another possibility is that we could also add a check here to make sure we are not pushing binary files.

Reply:
The only binary files are the fonts from the mdbook theme. These should change rarely, if ever, and only with changes the to the version of `mdbook` we're using.

I think this will be fine, but the custom action does have a `single-commit` option that ensures the `gh-pages` branch has history wiped out automatically. I don't plan on turning this on for now, but if this becomes a problem in the future, we could switch to using that.

Reply:
That's fair. Thanks for checking.

Reply:
Should we branch protect that branch, so only actions can update it